boot/zephyr/Kconfig

```diff
@@ -663,10 +663,6 @@ config MULTITHREADING
 	default n if SOC_FAMILY_NRF
 	default y
 
-config LOG_IMMEDIATE
```

Comment:
why not use LOG_MODE_IMMEDIATE instead?

This is commit with was introduced this https://github.com/mcu-tools/mcuboot/commit/9a4946ce9a6d20e1710096cac5a7b476f85de65f for some expalanation.

Reply:
I have seen the above commit, however a later commit e75e33d reverted the change by defining logging mode in prj.conf. It felt like a good time to clean it up.

Having the default and setting the Kconfig option in prj.conf is of course not the same but
- logging in Zephyr is undergoing refactoring so the settings are not really stable at the moment.
- LOG_MODE_IMMEDIATE is now a choice option, cannot be assigned a default value, the logic would need to be reworked.
- if we are not choosing the synchronous logging mode then the default in Zephyr is the deferred mode. It is however quite expensive, requires extra ~4K of flash.

Reply:
Ok. Will merge this then.
I prefer to configure as much as possible automatically for the developer (prj.conf is not automatic thing), but in that state of things it doesn't make sense as you elaborated.